smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptJmesPathVisitor.java

```diff
@@ -0,0 +1,315 @@
+        this.writer = writer;
+        this.accessor = accessor;
+        this.executionContext = accessor;
+        this.jmesExpression = expression;
+        scopeCount = 0;
+    }
+
+    private String makeNewScope(String prefix) {
+        scopeCount += 1;
+        return prefix + scopeCount;
+    }
+
+    private static boolean isNumeric(String strNum) {
+        if (strNum == null) {
+            return false;
+        }
+        try {
+            Double.parseDouble(strNum);
+        } catch (NumberFormatException nfe) {
+            return false;
+        }
+        return true;
+    }
+
+    public void run() {
+        writer.openBlock("let returnComparer = () => {", "}", () -> {
+            executionContext = this.accessor;
+            jmesExpression.accept(this);
+            writer.write("return $L;", this.executionContext);
+        });
+        executionContext = "returnComparer()";
+    }
+
+
+    void writeBooleanExpectation(String expectedValue, String returnValue) {
+        writer.openBlock("if ($L == $L) {", "}", this.executionContext, expectedValue, () -> {
+            writer.write("return $L", returnValue);
+        });
+    }
+
+    void writeAnyStringEqualsExpectation(String expectedValue, String returnValue) {
+        String element = makeNewScope("element");
+        writer.openBlock("for (let $L of $L) {", "}", element, this.executionContext, () -> {
+            writer.openBlock("if ($L == $L) {", "}", element, expectedValue, () -> {
+                writer.write("return $L;", returnValue);
+            });
+        });
+    }
+
+    void writeAllStringEqualsExpectation(String expectedValue, String returnValue) {
+        String element = makeNewScope("element");
+        String result = makeNewScope("result");
+        writer.write("let $L = false;", result);
+        writer.openBlock("for (let $L of $L) {", "}", element, this.executionContext, () -> {
+            writer.write("$L = $L || ($L == $L)", result, result, element, expectedValue);
+        });
+        writer.openBlock("if ($L) {", "}", result, () -> {
+            writer.write("return $L;", returnValue);
+        });
+    }
+
+    void writeStringExpectation(String expectedValue, String returnValue) {
+        writer.openBlock("if ($L === $S) {", "}", this.executionContext, expectedValue, () -> {
+            writer.write("return $L", returnValue);
+        });
+    }
+
+    @Override
+    public Void visitComparator(ComparatorExpression expression) {
+
+        String executionContextInital = this.executionContext;
+
+        expression.getLeft().accept(this);
+        String leftContext = this.executionContext;
+
+        this.executionContext = executionContextInital;
+
+        expression.getRight().accept(this);
+        String rightContext = this.executionContext;
+
+        this.executionContext = "(" + leftContext + " " + expression.getComparator().toString()
```

Comment:
What is executionContext exactly? Might be worth documenting that using javadocs. I thought it was the name of a variable, but it looks like it can be more than that.

Reply:
added